configsuite/schema.py

```diff
@@ -65,13 +68,19 @@ def _check_default_not_required(schema_level):
     return True
 
 
+@configsuite.validator_msg("Required can not have Default")
+def _check_required_not_default(schema_level):
+    return not schema_level.get(MK.Required, True) or MK.Default not in schema_level
```

Comment:
How does this differ from ` _check_default_not_required`? Looks to do the same. Except that the ` _check_default_not_required` has an unecessary check for `BasicType`.

In my opinion the ` _check_default_not_required` does too much, while `_check_required_not_default` could be more readable.

How about:
```
if schema_level.get(MK.Required, True):
     return MK.Default not in schema_level
return True
```

Reply:
Well spotted 🤦‍♂ Updated the PR now...